Sources/Extensions/SwiftStdlib/RangeReplaceableCollectionExtensions.swift

```diff
@@ -91,12 +87,64 @@ extension RangeReplaceableCollection {
         return remove(at: index)
     }
 
-    #if canImport(Foundation)
     /// SwifterSwift: Remove a random value from the collection.
     @discardableResult public mutating func removeRandomElement() -> Element? {
         guard let randomIndex = indices.randomElement() else { return nil }
         return remove(at: randomIndex)
     }
-    #endif
+
+    /// SwifterSwift: Keep elements of Array while condition is true.
+    ///
+    ///        [0, 2, 4, 7].keep(while: { $0 % 2 == 0 }) -> [0, 2, 4]
+    ///
+    /// - Parameter condition: condition to evaluate each element against.
+    /// - Returns: self after applying provided condition.
+    /// - Throws: provided condition exception.
+    @discardableResult
+    public mutating func keep(while condition: (Element) throws -> Bool) rethrows -> Self {
+        var idx = startIndex
+        for element in self {
+            if try !condition(element) {
+                removeSubrange(idx...)
+                break
+            }
+            formIndex(after: &idx)
+        }
+        return self
+    }
+
+    /// SwifterSwift: Take element of Array while condition is true.
+    ///
+    ///        [0, 2, 4, 7, 6, 8].take( where: {$0 % 2 == 0}) -> [0, 2, 4]
+    ///
+    /// - Parameter condition: condition to evaluate each element against.
+    /// - Returns: All elements up until condition evaluates to false.
+    public func take(while condition: (Element) throws -> Bool) rethrows -> Self {
```

Comment:
This is basically `prefix(while:)` and can be simplified with:
```Swift
public func take(while condition: (Element) throws -> Bool) rethrows -> Self {
    return Self(try prefix(while: condition))
}
```

Reply:
You're right, so maybe we should deprecate it in the future :)